src/libaktualizr/primary/aktualizr_test.cc

```diff
@@ -1082,6 +1082,47 @@ TEST(Aktualizr, InstallationFailure) {
   }
 }
 
+TEST(Aktualizr, SecondaryMetaFailure) {
+  TemporaryDirectory temp_dir;
+  auto http_server_mock = std::make_shared<HttpFake>(temp_dir.Path(), "hasupdates", fake_meta_dir);
+  Config conf = UptaneTestCommon::makeTestConfig(temp_dir, http_server_mock->tls_server);
+  auto storage = INvStorage::newStorage(conf.storage);
+
+  fault_injection_init();
+  fiu_enable("secondary_putmetadata", 1, nullptr, 0);
+
+  UptaneTestCommon::TestAktualizr aktualizr(conf, storage, http_server_mock);
+
+  struct {
+    result::Install result;
+    std::future<void> future;
+    std::promise<void> promise;
+  } ev_state;
+  ev_state.future = ev_state.promise.get_future();
+
+  auto f_cb = [&ev_state](const std::shared_ptr<event::BaseEvent>& event) {
+    if (event->variant != "AllInstallsComplete") {
+      return;
+    }
+    const auto installs_complete = dynamic_cast<event::AllInstallsComplete*>(event.get());
+    ev_state.result = installs_complete->result;
+
+    ev_state.promise.set_value();
+  };
+  boost::signals2::connection conn = aktualizr.SetSignalHandler(f_cb);
+  aktualizr.Initialize();
+  aktualizr.UptaneCycle();
+
+  auto status = ev_state.future.wait_for(std::chrono::seconds(20));
```

Comment:
it's really minor, just IMHO it would look slightly more readable if ev_state.future field is not defined at all and just `ev_state.promise.get_future().wait_for()` is used.